src/lib.rs

```diff
@@ -76,6 +99,22 @@ pub trait CompileConst {
             self.const_val()
         )
     }
+    /// Takes 3 strings: Attrbibutes, a visibility (eg pub) and a name
+    /// (a SCREAMING_SNAKE_CASE string is preferred) to use as the static's name,
+    /// then calls self.const_type() and self.const_val() in order to generate a
+    /// Rust compile-time static declaration statement.
+    fn static_declaration(&self, attrs: &str, vis: &str, name: &str) -> String {
+        format!(
+            "{}{}{}{}static {}: {} = {};",
+            if attrs.is_empty() { "" } else { attrs },
+            if attrs.is_empty() { "" } else { " " },
+            vis,
+            if vis.is_empty() { "" } else { " " },
+            name,
+            Self::const_type(),
+            self.const_val()
+        )
+    }
```

Comment:
@Eolu I have choosen here to duplicate the method from `const_declaration`, to make the change "minimally invasive" and backwards-compatible.

IMHO it would be nicer if we had a more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) approach here. Let me know if you thinks it's worth refactoring.

Reply:
Main concern here would be if we did something like this:

```rust
     fn declaration(&self, attrs: &str, vis: &str, as_static: bool, name: &str) -> String {
        format!(
            "{}{}{}{}{} {}: {} = {};",
            if attrs.is_empty() { "" } else { attrs },
            if attrs.is_empty() { "" } else { " " },
            vis,
            if vis.is_empty() { "" } else { " " },
            if as_static {"static"} else {"const"},
            name,
            Self::const_type(),
            self.const_val()
        )
    }

    fn const_declaration(&self, attrs: &str, vis: &str, name: &str) -> String {
        self.declaration(attrs, vis, false, name)
    }

    fn static_declaration(&self, attrs: &str, vis: &str, name: &str) -> String {
        self.declaration(attrs, vis, true, name)
    }
```

that it's not obvious for users who manually implemented `CompileConst` for a type that they would need to move their potentially custom implementation of  `const_declaration` to `declaration`

Reply:
I think the DRY approach is better here. Those that manually implemented `const_declaration` wouldn't be broken by this change, but would want to migrate to implementing `declaration` if they want to make use of both `const_declaration` and `static_declaration`. I'd add something to the documentation for `const_declaration` and `static_declaration` to make it clear that they're wrappers for `declaration`, which is the correct one to implement.

I also would prefer an enum in place of `as_static` to make it clear that we're choosing between `const` and `static`, something like this:
```rust
pub enum DeclarationType
{
    Const,
    Static
}
```

Reply:
Ok, I updated the PR accordingly 👍

Reply:
Looks good!